### PR TITLE
Master

### PR DESCRIPTION
**Pull request by yaoyuan7**

### CODE REVIEW

**Comment by yaoyuan7** [comment]
change something

File: Localization/autocommand.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+0,0
```
Comment (duembgen) [inline Localization/autocommand.txt]: what is the use of this file? Please delete if not used, or put inside a "inputs" folder.

File: Localization/drive.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+##@package move
+# move
+# ==============
+# Contains Robot class
+#
+#created by Frederike Duembgen, July 2015
```
Comment (duembgen) [inline Localization/drive.py]: Update the details.

File: Localization/drive.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        if (command[1] == ' ') and (command[3] == ' '): # spaces after command and board
+            if valid_boards.count(command[2]) > 0: # board is valid
+                if command[0] == 'g':
+                    #Â get mode
+                    print('get mode')
+                elif command[0] == 'p':
+                    #Â set mode
+                    print('set mode')
+    return 1
+def read_file(R,inputfile):
```
Comment (duembgen) [inline Localization/drive.py]: I think this function is not used. Remove it (and other unnecessary functions linked with it)

File: Localization/drive.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        command = 'i'
+        position = dict()
+        motors=["l","r"]
+        position[motors[0]]=-1
+        position[motors[1]]=-1
+        t0 = time.time()
+        
+        command0 = input('Drive mode begins, please enter your command:')
+        while command0 != 'end':
+            if (command0 != 'f' and command0 != 'b' and command0 != 'l' and command0 != 'r' and command0 != 'end' and command0 != 's'):
```
Comment (duembgen) [inline Localization/drive.py]: use valid_commands array for better readibility.

File: Localization/drive.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        print("running in debug mode")
+    else:
+        print("running in real mode")
+
+    if not DEBUG:
+        R.connect()
+
+    print("starting new movement")
+
+    R.move(output_result)
```
Comment (duembgen) [inline Localization/drive.py]: What happens to this in debug mode? Doesn't it throw an error?

File: Localization/get_image.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 # Simple module to read image from specified webcam.
 
 from __future__ import print_function
-import urllib
+import urllib.request
```
Comment (duembgen) [inline Localization/get_image.py:7]: Did you change this? Could you explain why?

File: Localization/inputs.txt [hunk 7]
```diff
@@ -0,0 +1,7 @@
+0	s
```
Comment (duembgen) [inline Localization/inputs.txt]: Same as above, put into input files folder.

File: Localization/newrun.py [hunk 8]
```diff
@@ -0,0 +1,313 @@
+# -*- coding: utf-8 -*-
```
Comment (duembgen) [inline Localization/newrun.py]: How much changed in this file with respect to the old move.py file? If not much changed, please overwrite move.py to avoid duplicates.

File: Localization/output/autodrive.ipynb [hunk 9]
```diff
@@ -0,0 +1,128 @@
+{
```
Comment (duembgen) [inline Localization/output/autodrive.ipynb]: Let's not add python notebooks to github, they are a pain in version control systems. Please add
the line *.ipynb to .gitignore file and remove these files from pull request.

File: Localization/output/drive.ipynb [hunk 10]
```diff
@@ -0,0 +1,151 @@
+{
```
Comment (duembgen) [inline Localization/output/drive.ipynb]: see comment above.

File: Localization/get_image.py (hunk 6 above)
Comment (yaoyuan7) [reply Localization/get_image.py:7]: because python3 use urllib.request library instead of urllib for requesting.

**Comment by yaoyuan7** [comment]
new pull request on FEB 15

**Comment by duembgen** [comment]
Thank you for implementing the changes! Please work on the target detection on a new branch, and change the title and description of this pull request to something more meaningful. 
I added more comments below. In particular, remove the python notebooks from the pull request (for the future, add the lines *.ipynb and .ipynb-checkpoints to .gitignore. Manually remove the notebooks from this pull request, see http://stackoverflow.com/questions/1274057/how-to-make-git-forget-about-a-file-that-was-tracked-but-is-now-in-gitignore).

File: Localization/requirements.txt [hunk 11]
```diff
@@ -1,9 +0,0 @@
-Requirements
```
Comment (duembgen) [inline Localization/requirements.txt:1]: Please don't remove this file.

File: Localization/output/move_analyse.ipynb [hunk 12]
```diff
@@ -0,0 +1,151 @@
+{
```
Comment (duembgen) [inline Localization/output/move_analyse.ipynb:1]: Please don't add python notebooks to the pull request.

File: Localization/output/autodrive_analyse.ipynb [hunk 13]
```diff
@@ -0,0 +1,128 @@
+{
```
Comment (duembgen) [inline Localization/output/autodrive_analyse.ipynb:1]: Please don't add python notebooks to the pull request.

File: Localization/output/.ipynb_checkpoints/move_analyse-checkpoint.ipynb [hunk 14]
```diff
@@ -0,0 +1,151 @@
+{
```
Comment (duembgen) [inline Localization/output/.ipynb_checkpoints/move_analyse-checkpoint.ipynb:1]: Add .ipynb_checkpoints in the .gitignore.

File: Localization/location.py [hunk 15]
```diff
@@ -355,6 +355,17 @@ def signal_handler(signal, frame):
             calib.write_pos(out_dir,name,np.matrix(np.array(np.array(r_real)[0]).reshape((1,3))))
 
 #--------------------------- 4.1 Visual localization ----------------------#
```
Comment (duembgen) [inline Localization/location.py:357]: Why are these lines moved? If this is on purpose, please also move the comments.

File: Localization/get_image.py (hunk 6 above)
Comment (duembgen) [reply Localization/get_image.py:7]: Ok!

File: Localization/autodrive.py [hunk 16]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (duembgen) [inline Localization/autodrive.py:1]: Please add ashort description of what this module does.
